plugins/org.eclipse.glsp.api/src/org/eclipse/glsp/api/operation/Operation.java

```diff
@@ -0,0 +1,53 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.api.operation;
+
+import org.eclipse.glsp.api.action.Action;
+
+/**
+ *
+ * An Operation is an action that directly manipulates the model representation on server side.
+ *
+ */
+public abstract class Operation extends Action {
+
+   public Operation(final String operationKind) {
+      super(operationKind);
+   }
+
+   @Override
+   public String toString() {
+      StringBuilder builder = new StringBuilder();
+      builder.append("Operation [kind=");
+      builder.append(getKind());
+      builder.append("]");
+      return builder.toString();
+   }
+
+   public static class Kind {
+      public static final String CREATE_CONNECTION = "createConnection";
```

Comment:
Shouldn't we rename this to CREATE_EDGE and RECONNECT_EDGE?

Reply:
I think the initial idea was to keep this as generic as possible. Depending on the diagram type not all connections are "edges" per se. (e.g. flows, transitions, dependency ...)
But I'm also fine with renaming this to reduce ambiguousness

Reply:
Hm, to me edge is a graphical representation ("line") of anything including flows, transitions, dependency, generalizations. In that sense all diagrams consist of nodes and edges. So I'm in favor of calling it edge.